src/components/BaseCard.vue

```diff
@@ -0,0 +1,15 @@
+<template>
```

Comment:
All the logic for the base card should go into this component, there is no need to use slots, just use v-for from outside and send the data using props